src/server/index.js

```diff
@@ -109,96 +115,219 @@ const createStock = quote => {
   };
 };
 
-const calculateDateYearsInPast = years => {
-  const year = new Date().getFullYear() - years;
-  const date = new Date();
-  date.setFullYear(year);
-  return date;
+// Used for historical() data obtained using period 'd'
+const getDatesAndPrices = dailyData => {
+  let datesAndPrices = [];
+  dailyData.forEach(({
+    date,
+    close
+  }) => {
+    datesAndPrices.unshift({
+      date: formatDateForMaxStockData(date),
+      price: close
+    });
+  });
+  return datesAndPrices;
+};
+
+// NOTE: this date has timezone UTC, which is incorrect but works in this
+// case because we are just extracting the time
+const getAdjustedDateForTimestamp = (gmtoffset, timestamp) => {
+  const adjustedTimestamp =
+    (timestamp + gmtoffset) * constants.MILLISECONDS_PER_SECOND;
+  return new Date(adjustedTimestamp);
 };
 
-const padSingleDigitWithZero = value => {
-  let num = parseInt(value);
-  // need to check value because parseInt turns '12hello' into a number
-  if (isNaN(value) || isNaN(num)) {
-    throw new TypeError(`${padSingleDigitWithZero.name} requires a number or numeric string`);
+const getDateAndTime = (gmtoffset, timestamp, dateAndTimeFormat) => {
+  const dateAndTime = getAdjustedDateForTimestamp(gmtoffset, timestamp);
+  return dateFormat(dateAndTime, dateAndTimeFormat, true);
+};
+
+const getStartOfDayTimestampIndex = (dayIndex, timestampsPerDay) =>
+  Math.floor(dayIndex * timestampsPerDay);
+
+const getEndOfDayTimestampIndex = (dayIndex, timestampsPerDay) =>
+  Math.floor((dayIndex + 1) * timestampsPerDay);
+
+// Could be buggy depending on how the timestamps in their API are given.
```

Comment:
I'm guessing timestamps are pretty consistent across JS stuff (the number of milliseconds elapsed since January 1, 1970 00:00:00 UTC), but yeah not positive

Reply:
I meant as long as it gives 5 days worth of timestamps to the beginning of whatever day is 5 days ago no matter what time of day you use the API. Do the timestamps go back to the start of 5 days ago the same way if you check at 12:15 am as opposed to 11:45 pm? I'm not sure because it's the weekend so I can't test it yet